core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1239,7 +1233,7 @@ public SearchSourceBuilder readFrom(StreamInput in) throws IOException {
             builder.queryBuilder = in.readQuery();
         }
         if (in.readBoolean()) {
-            int size = in.readVInt();
+            size = in.readVInt();
```

Comment:
This also seems to let `org.elasticsearch.messy.tests.SearchStatsTests.testSimpleStats` fail consistently for me when running locally (`Dtests.seed=A8EFE9FB672752E6`)